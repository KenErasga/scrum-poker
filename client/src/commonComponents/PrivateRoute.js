import React, {useContext} from 'react';
import { Redirect, Route } from 'react-router-dom';
import { AuthContext } from '../components/Accounts/CognitoProvider';

const PrivateRoute = ({ component: Component, ...rest }) => {
  const { isAuthenticated } = useContext(AuthContext);
  return (
    <Route
      {...rest}
      render={props =>
        isAuthenticated ? (
          <Component {...props} />
        ) : (
          <Redirect to={{ pathname: '/', state: { from: props.location } }} />
        )
      }
    />
  )
};

export default PrivateRoute