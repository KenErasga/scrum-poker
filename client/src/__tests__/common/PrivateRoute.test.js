import React from 'react';
import renderer from 'react-test-renderer';
import PrivateRoute from '../../common/PrivateRoute';
import {Router, MemoryRouter} from 'react-router-dom';

jest.mock('../../components/Error/ErrorHandler.js', () => ({
    useErrorHandler: () => ({
        isError: false,
        errorMessage: 'test error message',
        setIsError: jest.fn(),
        setErrorMessage: jest.fn(),
        socketError: 'socket error',
        disconnectError: 'disconnect error'
    })
})); 

describe('PrivateRoute', () => {

    it('PrivateRoute renders isAuthenticated=true correctly', () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                <PrivateRoute path='/test' component={() => <h2>test1</h2>} isAuthenticated={true}/>
            </MemoryRouter>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
    });

    it('PrivateRoute renders when isAuthenticated=false correctly', () => {
        const tree = renderer
            .create(
                <MemoryRouter>
                    <PrivateRoute path='/t' component={() => <h2>test2</h2>} isAuthenticated={false}/>
                </MemoryRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
        });
    
});
