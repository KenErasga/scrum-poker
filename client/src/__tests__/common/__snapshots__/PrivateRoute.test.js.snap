// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute PrivateRoute renders isAuthenticated=true correctly 1`] = `null`;

exports[`PrivateRoute PrivateRoute renders when isAuthenticated=false correctly 1`] = `null`;
